features/atcryptoauth/atca_app.cpp

```diff
@@ -22,42 +22,48 @@
 
 #if MBED_CONF_ATCAECC_APP_ENABLE
 
-/** ATCAECC508A sample config zone data. Required for calculating CRC while
- *  locking the config zone.
+/** Sample configuration data. It is supplied to the Lock command to
+ *  match the device configuration up to a CRC before locking. Hence it
+ *  protects against most accidental errors.
+ *
+ *  Note this data is deliberately not constant. First 4 blocks of 4 bytes
+ *  contain device specific data that changes from device to device. It is
+ *  replaced by reading these blocks from the device before calculating the
+ *  CRC.
  */
 uint8_t atcaecc508a_sample_config[] = {
-    0x01, 0x23, 0x95, 0x11,
-    0x00, 0x00, 0x50, 0x00,
-    0x5D, 0xCF, 0xD6, 0xCD,
-    0xEE, 0xC0, 0x1D, 0x00,
-    0xC0, 0x00, 0x55, 0x00,
-    0x83, 0x20, 0x83, 0x20,
-    0x83, 0x20, 0x83, 0x20,
-    0x83, 0x20, 0x83, 0x20,
-    0x83, 0x20, 0x83, 0x20,
-    0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0xAF, 0x8F,
-    0xFF, 0xFF, 0xFF, 0xFF,
-    0x00, 0x00, 0x00, 0x00,
-    0xFF, 0xFF, 0xFF, 0xFF,
-    0x00, 0x00, 0x00, 0x00,
-    0xFF, 0xFF, 0xFF, 0xFF,
-    0xFF, 0xFF, 0xFF, 0xFF,
-    0xFF, 0xFF, 0xFF, 0xFF,
-    0xFF, 0xFF, 0xFF, 0xFF,
-    0x00, 0x00, 0x55, 0x55,
-    0xFF, 0xFF, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00,
-    0x13, 0x00, 0x13, 0x00,
-    0x13, 0x00, 0x13, 0x00,
-    0x13, 0x00, 0x13, 0x00,
-    0x13, 0x00, 0x13, 0x00,
-    0x1C, 0x00, 0x10, 0x00,
-    0x10, 0x00, 0x10, 0x00,
-    0x10, 0x00, 0x10, 0x00,
-    0x10, 0x00, 0x1C, 0x00
+   0x01, 0x23, 0x95, 0x11,
```

Comment:
Since the first 16 bytes are a unique identifier, it would be better to make them a distinctive pattern like all-zeros than something that works on one particular device (and could mess up testing when we use that particular device for testing).